Rpg npc/struct features

anshikab opened the pull request:
This branch includes additions to the NPC struct that accommodate rpg related features. Namely, it integrates the playerstats struct, stats_t (implemented by the rpg_playerstats team here: https://github.com/uchicago-cs/chiventure/blob/rpg-stats/include/game-state/stats.h) into the NPC module. NPC related functions have been updated to reflect the change, as well as tests. The tests have been updated to include an empty stats_t struct as input. The relevant files are npc.c, npc.h and test_npc.c
If approved and merged successfully, this pull request Closes #600

firatciftci commented on src/game-state/src/npc.c:
```diff
@@ -0,0 +1,115 @@
+    if ((npc->health + change) < max)
+    {
+        npc->health += change;
+    }
```
I think the function header comment for `change_npc_health` should mention that the health change has a lower floor of `0` and an upper floor of `max`. Also, I'm not sure why the `max` value is being passed in as a parameter to this function. Wouldn't it make more sense to have a `max_health` numeric variable in the NPC struct itself, rather than managing it via the change function?

philipchenaux commented on src/game-state/src/npc.c:
```diff
@@ -0,0 +1,115 @@
+}
+
+/* See npc.h */
+int npc_free(npc_t *npc)
+{
+    assert(npc != NULL);
+    
+    // missing free_dialog function 
+    free(npc->npc_id);
+    delete_all_items(&npc->inventory);
+    free(npc);
```
I'm not sure if this is an issue, but do you need to free the stats struct as well? I always thought that you have to free the items inside the struct as well, before freeing the struct as a whole.

philipchenaux commented on tests/game-state/test_npc.c:
```diff
@@ -0,0 +1,100 @@
+Test (npc, new) 
+{
+	npc_t *npc; 
+	npc = npc_new("npc_22", 20, NULL, NULL);
```
I guess you can make this a separate issue to create a sample testing environment, but maybe passing in a sample stat into this test could help? You can call `stat_new` and create a sample stat that you will then pass into the test.

philipchenaux commented on tests/game-state/test_npc.c:
```diff
@@ -0,0 +1,100 @@
+Test (npc, init) 
+{ 
+	npc_t *npc; 
+	int init = npc_init(&npc, "npc_22", 20, NULL, NULL); 
```
Similar to the comment above, the same could apply here.

anshikab commented:
I have committed changes to the code in the rpg-npc/struct-features branch as per the comments provided:
- I have updates the header of the npc_change_health function to reflect the minimum and maximum values of the change. I have not, however, changed the implementation as this function was written by the npc team.
- I have modified npc_free to free the stats_t struct as well.
- As per recent conversations with the npc team, I have added these modifications to files in the new npc module, instead of those (now outdated) in the game-state module.
- The tests still only take empty stats_t structs; I that creating a separate issue to come up with a new testing environment stats would work better.

firatciftci commented:
Issue Score: ✔️+

Comments:
Great work! The only thing I'll point out is that the title of the PR should have been more detailed; ideally, it should be a short sentence about the main change coming with the PR.

<!-- {"score": "check-plus"} -->
